github/ci/services/cert-manager/hack/common.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -eo pipefail
+
+TOTAL_WAIT_TIME=40
+NEXT_WAIT_TIME=0
+
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+RED='\033[0;31m'
+NOCOLOR='\033[0m'
+
+log(){
+    local msg="${1}"
+    local color="${2}"
+
+    echo -e "${color}${msg}${NOCOLOR}"
+}
+
+log_error(){
+    local msg="${1}"
+
+    log "${msg}" "${RED}"
+}
+
+log_success(){
+    local msg="${1}"
+
+    log "${msg}" "${GREEN}"
+}
+
+log_info(){
+    local msg="${1}"
+
+    log "${msg}" "${YELLOW}"
+}
+
+install_kubectl(){
+    log_info "Installing kubectl"
+
+    curl -LO "https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl"
+    chmod a+x ./kubectl && mv ./kubectl /usr/local/bin/kubectl
+}
+
+wait_http01_ingress_created(){
+    local namespace=${1:-default}
+    local label_selector="acme.cert-manager.io/http01-solver=true"
+    local expected_path=".well-known/acme-challenge/"
+
+    if ! which kubectl; then
+        install_kubectl
+    fi
+
+    log_info "Checking http01 ingress in namespace ${namespace}..."
+
+    until [ "$(kubectl get ingress -n ${namespace} -l ${label_selector} -o yaml | grep ${expected_path})" ] || [ $NEXT_WAIT_TIME -eq $TOTAL_WAIT_TIME ]; do
+        log_info "http01 ingress in namespace ${namespace} not created yet..."
+        sleep $(( NEXT_WAIT_TIME++ ))
+    done
+
+    if [ $NEXT_WAIT_TIME -eq $TOTAL_WAIT_TIME ]; then
+        log_error "http01 ingress in namespace ${namespace} not created in time, exiting"
+        exit 1
+    fi
+    log_success "http01 ingress in namespace ${namespace} found"
+}
+
+wait_pod_ready(){
+    local pod=$1
+    local namespace=${2:-default}
+    local target=${3:-"1/1"}
+
+    if ! which kubectl; then
+        install_kubectl
```

Comment:
Too bad that the gitops rules are not exposing their internal `kubectl` target as an executable rule. Something we could probably create an issue on their repo.